### PR TITLE
Add test for retrying of submission completion when queries fail

### PR DESCRIPTION
**Pull request by ronaldfw**
We recently added retrying completing a submission upon failure of
timestamp queries in the Vulkan layer. Here we add a unit test that checks for the retry
on query fail.

Tested: Ran unit test (only unit test code was changed).
Bug: http://b/179649352

### CODE REVIEW

File: src/OrbitVulkanLayer/SubmissionTrackerTest.cpp [hunk 1]
```diff
@@ -536,6 +538,99 @@ TEST_F(SubmissionTrackerTest,
...
+  post_submit_times[1] = orbit_base::CaptureTimestampNs();
+
+  tracker_.CompleteSubmits(device_);
+  tracker_.CompleteSubmits(device_);
+
+  EXPECT_THAT(actual_reset_slots1, UnorderedElementsAre(kSlotIndex1, kSlotIndex2, kSlotIndex4));
+  // We failed on the 4th GetQuerySlots call above, which belongs to the "begin" timestamp of the
+  // second command buffer (note that the order of calls when querying for timestamps is to query
+  // for the end timestamp first). Therefore, this must be kSlotIndex3.
+  // TODO: Consider if we should just check the size of the reset slots? Depending on code order
```
Comment (ronaldfw) [inline src/OrbitVulkanLayer/SubmissionTrackerTest.cpp:624]: Thoughts?

Comment (florian-kuebler) [reply src/OrbitVulkanLayer/SubmissionTrackerTest.cpp:624]: I was also thinking about this. I guess it is fine for the tests. We had the tests from quite early on, and changed the code and the tests quite a bit. AFAIR we did't had issues with slots indices. So I think this is fine for the tests. Alternative would be to collect the slots into the same vector, and just check that all of them are reset exactly once.

**Review by florian-kuebler — APPROVED** [review]
LGTM. Thanks!

File: src/OrbitVulkanLayer/SubmissionTrackerTest.cpp (hunk 1 above)
Comment (ronaldfw) [reply src/OrbitVulkanLayer/SubmissionTrackerTest.cpp]: Ok, if we find it bothers us, we can change it later. I'll merge as-is. Thanks for the input!

**Pull request merged**